docs/dev/guides/fragment-controllers.md

```diff
@@ -0,0 +1,162 @@
+
+From an HTTP perspective, each (sub) request is totally independent. Each one will get request
+parameters (like `GET` and `POST` data), and they can return HTTP headers (like `Content Type`
+or `Cache-Control`).
+
+{{% notice note %}}
+Since **Contao 4.9**, subrequests can generate [a response that modifies the cache time of
+the main response](/framework/caching/#caching-fragments).
+{{% /notice %}}
+
+
+
+## Fragment Controllers
+
+Contao provides two different types of fragments out-of-the-box:
+[Front End Modules][moduels] and [Content Elements][elements]. For each of
+these, there are abstract classes (namely `AbstractFrontendModuleController`
+and `AbstractContentElementController`) that mimick the behavior of a Contao 3
+module/element. They generate the Contao template based on the request and current
+model, and allow to
+
+
+### Controllers in the Contao 3 framework
+
+Front end modules and content elements are still generated by the Contao 3
+framework and registered in the `$GLOBALS['FE_MOD']` and `$GLOBALS['TL_CTE']`
+respectively. The `ModuleProxy` and `ContentProxy` classes are responsible
+for making a fragment controller behave like a Contao 3 module/element.
+
+
+### Extending a Contao 3 controller
+
+{{% notice tip %}}
+Modules are used to simplify, everything mentioned here works exactly the same for content elements.
+{{% /notice %}}
+
+Sometimes, one might need to override/extend the functionality of an existing Contao 3 module.
+Sure one could simply create a module *the old way*, but this limits functionality and is not
+recommended anymore. However, there's nothing limiting us from extending an existing class and
+using it as a controller in Contao 4!
+
+```php
+// src/Controller/FrontendModule/AppExampleController.php
+namespace App\Controller\FrontendModule;
+
+use Contao\CoreBundle\ServiceAnnotation\FrontendModule;
+use Contao\ModuleModel;
+use Contao\ModuleNewsList;
+use Symfony\Component\HttpFoundation\Response;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * @FrontendModule(category="news")
+ */
+class AppExampleController extends ModuleNewsList implements ServiceAnnotationInterface
+{
+    public function __construct() {}
+
+    public function __invoke(ModuleModel $model, string $section): Response
+    {
+        parent::__construct($model, $section);
+
+        return new Response($this->generate());
+    }
+    
+    // Do whatever you want here, e.g. override parent methods
+}
+```
+
+{{%expand "Code example to extend a content" %}}
+```php
+// src/Controller/ContentElement/AppExampleController.php
+namespace App\Controller\ContentElement;
+
+use Contao\ContentGallery;
+use Contao\ContentModel;
+use Contao\CoreBundle\ServiceAnnotation\FrontendModule;
+use Symfony\Component\HttpFoundation\Response;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * @FrontendModule(category="news")
```

Comment:
has to be CE annotation here

Reply:
fixed in 990eff98c449113f133e2811d7a406aeb7fa01f7